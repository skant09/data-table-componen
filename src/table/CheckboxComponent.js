import React, {useState} from 'react';


const CheckboxComponent = props => {
  const [hoverState, setHoverState] = useState(props.checked);
  const toggelCheckboxState = e => {
    setHoverState(!hoverState);
  }
  return (
    <input type="checkbox" value={props.id} checked={Boolean(props.checked)} onChange={toggelCheckboxState} />
  )
}
export default CheckboxComponent