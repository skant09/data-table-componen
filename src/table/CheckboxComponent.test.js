import React from 'react';
import {shallow} from 'enzyme';
import CheckboxComponent from './CheckboxComponent';
import renderer from 'react-test-renderer';
import columnsConfig from './index'

const mockedEventhandler = jest.fn(event => {
  console.log("Mocked function");
});

it('renders without crashing', () => {
  shallow(<CheckboxComponent />);
});

it('matches snapshot', () => {
  const app = renderer.create(<CheckboxComponent onChnage={mockedEventhandler} columns={columnsConfig} rows={[]}/>).toJSON();
  expect(app).toMatchSnapshot();
});
