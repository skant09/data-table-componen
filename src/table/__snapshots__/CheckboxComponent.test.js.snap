// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<input
  checked={false}
  onChange={[Function]}
  type="checkbox"
/>
`;
